tabular/src/autogluon/tabular/learner/abstract_learner.py

```diff
@@ -170,30 +170,26 @@ def transform_features(self, X):
     def score(self, X: DataFrame, y=None, model=None):
         if y is None:
             X, y = self.extract_label(X)
-        X = self.transform_features(X)
-        y = self.label_cleaner.transform(y)
-        trainer = self.load_trainer()
-        if self.problem_type == MULTICLASS:
-            y = y.fillna(-1)
-            if (not trainer.eval_metric_expects_y_pred) and (-1 in y.unique()):
-                # log_loss / pac_score
-                raise ValueError(f'Multiclass scoring with eval_metric=\'{self.eval_metric.name}\' does not support unknown classes.')
-        return trainer.score(X=X, y=y, model=model)
+        self._validate_class_labels(y)
+        if self.eval_metric.needs_pred:
+            y_pred = self.predict(X=X, model=model)
+            return self.eval_metric(y, y_pred)
+        else:
+            y_pred_proba = self.predict_proba(X=X, model=model)
+            return self.eval_metric(y, y_pred_proba)
 
     # Scores both learner and all individual models, along with computing the optimal ensemble score + weights (oracle)
     def score_debug(self, X: DataFrame, y=None, extra_info=False, compute_oracle=False, silent=False):
         leaderboard_df = self.leaderboard(extra_info=extra_info, silent=silent)
         if y is None:
             X, y = self.extract_label(X)
+        self._validate_class_labels(y)
+
         X = self.transform_features(X)
-        y = self.label_cleaner.transform(y)
-        trainer = self.load_trainer()
-        if self.problem_type == MULTICLASS:
-            y = y.fillna(-1)
-            if (not trainer.eval_metric_expects_y_pred) and (-1 in y.unique()):
-                # log_loss / pac_score
-                raise ValueError(f'Multiclass scoring with eval_metric=\'{self.eval_metric.name}\' does not support unknown classes.')
+        y_internal = self.label_cleaner.transform(y)
+        y_internal = y_internal.fillna(-1)
```

Comment:
doesn't balanced_accuracy bug still exist in score_debug() if two different unknown classes are replaced by -1?

Reply:
No, because y_internal is only used in pred_proba metrics, but balanced_accuracy is a pred metric.